### PR TITLE
T-23 : Package Signing in Package Creator Tool

### PR DESCRIPTION
**Pull request by ozdemirfadime**

### CODE REVIEW

File: tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs [hunk 1]
```diff
@@ -55,7 +55,8 @@ public byte[] Encrypt(byte[] plainData)
```
Comment (zaya-mc) [inline tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs:55]: @ozdemirfadime @solakogullarielif how do we pass Private key? I could not see you are passing the key.

**Review by zaya-mc — CHANGES_REQUESTED** [review]

File: tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs [hunk 2]
```diff
@@ -93,34 +94,22 @@ public byte[] Sign(byte[] data)
...
                 }
             }
 
             catch (CryptographicException e)
             {
                 Console.WriteLine(e.Message);
             }
+            			
+			/* Remove the below line once you return a proper signature */
+			return  SignedHash;
```
Comment (ozdemirfadime) [inline tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs:109]: You can remove this return line so you already returned SignedHash

File: tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs (hunk 1 above)
Comment (ozdemirfadime) [reply tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs:55]: we committed only last review so we will one more commit that contain a private key

File: source/boot_security.c [hunk 3]
```diff
@@ -22,24 +22,49 @@
...
+
+/*
+ * Authentication Key Len
+ * We use ECC P256 for authentication.
+ * The public key len is 2 key len : 2 x 32
+ */
+#define AUTHENTICATION_KEY_LEN_IN_BYTES (64)
+
+
+
```
Comment (zaya-mc) [inline source/boot_security.c]: My mistake but could you please remove two empty lines.

File: tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs [hunk 4]
```diff
@@ -2,8 +2,10 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Runtime.CompilerServices;
```
Comment (zaya-mc) [inline tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs]: We should not need that, please remove

File: tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs [hunk 5]
```diff
@@ -2,8 +2,10 @@
...
 using System.Security.Cryptography;
 using System.Text;
+using System.Windows.Forms;
```
Comment (zaya-mc) [inline tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs]: We should not need that, please remove

**Pull request merged**